dagshub/data_engine/model/query_result.py

```diff
@@ -623,22 +694,63 @@ def visualize(self, visualizer: Literal["dagshub", "fiftyone"] = "dagshub", **kw
             return sess
 
     def annotate(
-        self, open_project=True, ignore_warning=True, fields_to_embed=None, fields_to_exclude=None
+        self,
+        open_project=True,
+        ignore_warning=True,
+        fields_to_embed=None,
+        fields_to_exclude=None,
+        model_name=None,
+        model_version="latest",
+        model_repo=None,
+        model_pre_hook=lambda x: x,
+        model_post_hook=None,
+        model_batch_size=1,
+        model_metadata_column="annotation",
+        use_remote_backend=False,
     ) -> Optional[str]:
         """
         Sends all the datapoints returned in this QueryResult to be annotated in Label Studio on DagsHub.
+        Alternatively, uses MLFlow to automatically label datapoints.
 
         Args:
             open_project: Automatically open the Label Studio project in the browser
             ignore_warning: Suppress the prompt-warning if you try to annotate too many datapoints at once.
             fields_to_embed: list of meta-data columns that will show up in Label Studio UI.
              if not specified all will be displayed.
             fields_to_exclude: list of meta-data columns that will not show up in Label Studio UI
+            model_repo: repository to extract the model from
+            model_name: name of the model in the mlflow registry
+            model_version: (optional, default: 'latest') version of the model in the mlflow registry
+            model_pre_hook: (optional, default: identity function) function that runs before datapoint is sent to the model
+            model_post_hook: function that converts mlflow model output converts to labelstudio format
+            model_batch_size: function that converts to labelstudio format
         Returns:
             The URL of the created Label Studio workspace
         """
         send_analytics_event("Client_DataEngine_SentToAnnotation", repo=self.datasource.source.repoApi)
 
+        if all([param is not None for param in [model_name, model_version, model_repo, model_post_hook]]):
+            if use_remote_backend:
+                # send to dagshub backend
+                raise NotImplementedError("Coming soon!")
```

Comment:
If this is not implemented then I don't see a point in having this is an argument. Delete it and we can come back to it once we need it